### DISCUSSION TITLE
Do we require any explicit dependency for fltk-rs project ?

### DISCUSSION THREAD

**Discussion by ApurvaSolanki54** [fltk-rs/fltk-rs#653] [Q&A] [answered]
Hi, I am creating Linux package i.e. .deb and .rpm for project that I have created using fltk-rs, I am using vlc-rs in that project, so do I have to specify any other dependency except libvlc for that project ?

**Comment by eadf** [comment, marked as answer]
To make fltk compile in GitHub-workflows ubuntu-latest I had to add this:
`apt-get install -y libpango1.0-dev libx11-dev libxext-dev libxft-dev libxinerama-dev libxcursor-dev libxrender-dev libxfixes-dev`
(fltk workflow uses this too). So I guess that's a minimum just to get it to compile.

Edit: the same packages minus the "-dev" would be required for a binary distribution

**Reply by ApurvaSolanki54** [reply]
Actually, I am directly including Compiled binary to .deb and .rpm package for others to download and install.

**Reply by eadf** [reply]
as I wrote.. libpango1.0 libx11 libxext libxft libxinerama libxcursor libxrender libxfixes dep. for a bin dist.

**Reply by MoAlyousef** [reply]
> Actually, I am directly including Compiled binary to .deb and .rpm package for others to download and install.

AFAIK for deb packages, you need to specify (not bundle) all runtime dependencies, meaning that even on a non-gui debian distro, running apt-get on your package should download everything needed to run the executable, including libx11 etc. Even though most gui distros would already have libx11 etc as part of the system libs necessary to run the gui distro, in which case apt-get won't re-download already installed libs. 
If your app doesn't use right-to-left or CJK fonts, you can build your app with the "no-pango" feature, and shave off some runtime dependencies.
